service/src/java/org/apache/hive/service/cli/operation/OperationManager.java

```diff
@@ -201,11 +202,28 @@ private void addOperation(Operation operation) {
     }
   }
 
+  public void updateQueryTag(String queryId, String queryTag) {
+    Operation operation = queryIdOperation.get(queryId);
+    if (operation != null) {
+      if (queryTagToIdMap.get(queryTag) != null) {
+        throw new RuntimeException("tag " + queryTag + " is already applied for query " + queryId);
+      }
+      queryTagToIdMap.put(queryTag, queryId);
+      LOG.debug("Query " + queryId + " is updated with tag " + queryTag);
```

Comment:
Can be info log as we can see which queryId mapped to what tag.

Reply:
done